api4/post.go

```diff
@@ -396,6 +397,54 @@ func getPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPostsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
+	var postIDs []string
+
+	json.NewDecoder(r.Body).Decode(&postIDs)
```

Comment:
Along those same lines, we might want to wrap `r.Body` in an `io.LimitReader` to avoid causing problems if someone sends an arbitrarily long array of IDs to the server. I can't remember if we have to do that every time we read arbitrary JSON from the request body, but I do remember doing that other places where we were using `ReadAll` to read image data and such

Reply:
1000 IDs shouldn't be more than about 35 KB. 

I do see `LimitReader` used in a few place and a lot of places that are using `ReadAll` unlimited, deferring to limits imposed by the underlying infrastructure. Another option is [`MaxBytesReader`](https://pkg.go.dev/net/http#MaxBytesReader). I'm 3/5 that this should be handle consistently across the API. 

Perhaps the Server Platform team (cc @agnivade @isacikgoz) have related plans? Perhaps a campaign is in order?

Reply:
This is already there: https://github.com/mattermost/mattermost-server/blob/df3ed307df0924f57363ca990703862d66193633/web/handlers.go#L151-L157.